include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,356 @@
+           * Takes as input the position and current time weight.
+           *
+           * @param position The current position to compute velocity
+           * @param time_weight A weighting between the two current timesteps
+           * n and n+1
+           */
+          Tensor<1,dim> surface_velocity(const Point<dim> &position,
+                                         const double time_weight) const;
+
+        private:
+
+         /**
+          * Tables which contain the velocities
+          */
+          dealii::Table<dim-1,Tensor<1,3> > velocity_vals;
+          dealii::Table<dim-1,Tensor<1,3> > old_velocity_vals;
+
+          /**
+           * Distances between adjacent point in the x/y grid
+           */
+          Point<dim-1> spacing;
+
+          /**
+           * Model size
+           */
+          Point<dim-1> grid_extent;
+
+          /**
+           * Number of points in the velocity grid
+           */
+          dealii::TableIndices<dim-1> npoints;
+
+
+          /**
+           * Dimensions of the boundary plane
+           */
+          unsigned int boundary_dimensions[dim-1];
+
+          /**
+           * Scale the velocity boundary condition by a scalar factor. Can be
+           * used to transform the unit of the velocities (if they are not
+           * specified in the default unit (m/s or m/yr depending on the
+           * "Use years in output instead of seconds" parameter).
+           */
+          const double scale_factor;
+
+
+          /**
+           * calculates the index given a certain position
+           *
+           * @param position Input position, which is converted to spatial
+           * index
+           */
+          dealii::TableIndices<dim-1>
+          calculate_spatial_index(const Point<dim> &position) const;
+
+          /**
+           * calculate the interpolation weight of a position
+           *
+           * @param point_weight Weight for the linear interpolation
+           */
+          void
+          calculate_point_weight(double *point_weight,
+                                 const Point<dim> &position,
+                                 const dealii::TableIndices<dim-1> spatial_index) const;
+
+          /**
+          * Convert a tensor of rank 1 and dimension in to rank 1 and
+          * dimension out. If $out < in$ the last elements will be discarded,
+          * if $out > in$ zeroes will be appended to fill the tensor.
+          */
+          template <int in, int out>
+          Tensor<1,out> convert_tensor (const Tensor<1,in> &old_tensor) const;
+
+
+          /**
+           * This function adds a certain data point to the interpolated
+           * surface velocity at this evaluation point.
+           */
+          Tensor<1,dim>
+          add_interpolation_point(const dealii::TableIndices<dim-1> spatial_index,
+                                  const double point_weight[dim-1],
+                                  const double time_weight) const;
```

Comment:
I think the function is misnamed. If you name a function `add_*`, that suggests that something is added to something. But what? The function only takes const arguments, and the function is const itself, so apparently nothing is modified. I suppose you mean `compute_interpolation_point` where the verb indicates that something is computed and returned?

You also probably want the first index to be passed by reference.